WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Settings/In-Person Payments/Deposits Overview/WooPaymentsDepositsCurrencyOverviewViewModelTests.swift

```diff
@@ -0,0 +1,57 @@
+import XCTest
+@testable import WooCommerce
+
+final class WooPaymentsDepositsCurrencyOverviewViewModelTests: XCTestCase {
+
+    var sut: WooPaymentsDepositsCurrencyOverviewViewModel!
+    var analyticsProvider: MockAnalyticsProvider!
+
+    override func setUp() {
+        analyticsProvider = MockAnalyticsProvider()
+        let analytics = WooAnalytics(analyticsProvider: analyticsProvider)
+        sut = WooPaymentsDepositsCurrencyOverviewViewModel(overview: .fake(), analytics: analytics)
+    }
+
+    func test_when_expand_is_tapped_analytic_event_is_tracked() {
+        // Given
+        let expanded = true
+
+        // When
+        sut.expandTapped(expanded: expanded)
+
+        // Then
+        XCTAssertTrue(analyticsProvider.receivedEvents.contains(WooAnalyticsStat.paymentsMenuDepositSummaryExpanded.rawValue))
+    }
+
+    func test_when_collapse_is_tapped_analytic_event_is_not_tracked() {
```

Comment:
Just wondering: 

For this specific test, would it make sense to instead of calling `expandTapped(false)` and checking that we have not received the event, to call the method twice (expand + collapse), and then assert the count of events to confirm it has been only logged once?

Reply:
Well, we'd need to call it with `true` first, then `false`, so it adds unneccesary complexity to the test. 

We don't have any logic in the view model about this (giving bindings to the VM is a pain, and the state isn't owned by the view or viewModel which changes it) so the method we're testing is _informing_ the view model about the expansion/collapse, not about a toggle event.